components/automate-ui/src/app/helpers/auth/regex.ts

```diff
@@ -23,7 +23,9 @@ export class Regex {
     NO_MIXED_WILDCARD_ALLOW_SPECIAL: '^(\\*|[^:*]+)$',
 
     // Allows valid FQDN only
-    VALID_FQDN: /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-zA-Z0-9_]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z]{2,5}(:[0-9]{1,5})?(\/.*)?$/,
+    // Top level domain is limited to a maximum number of 25 characters
+    VALID_FQDN: /^(https?:\/\/)?[a-zA-Z0-9_]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z]{2,25}(:[0-9]{1,5})?(\/.*)?$/,
```

Comment:
I ended up shortening just the first capture group.  When playing around with the sample you provided, I kept running into "Catastrophic Backtracing".  Even though this isn't a perfect Regex (as you demonstrated is nearly impossible). Its really a pretty good one for our needs.